### ISSUE TITLE
PEB Commandline

### ISSUE THREAD

**Issue by shainodev** [UnamSanctam/SilentCryptoMiner#2252]
How do u get the commandline from PEB in the xmrig? I want to use my own xmrig version but I cant figure out how you get the miner arguments from the peb

**Comment by shainodev** [comment]
Could you please check your protonmail?

**Comment by UnamSanctam** [comment]
>Could you please check your protonmail?

I think I've responded to all messages I've received.

**Comment by shainodev** [comment]
I'll reply to my message there, then you will see

**Comment by shainodev** [comment]
how do you use the CurProcessParameters->ShellInfo for Process process(argc, argv); on xmrig?

**Comment by shainodev** [comment]
nvm. got it to work, but now im wondering i switched to the moneroocean's xmrig, am i losing any crucial features that u have added? i dont mean the stealth targets or idle mining or any such thing i mean like a important bugfix in xmrig or something that decreases detections alot or am i not?

**Comment by shainodev** [comment]
also how long does the miners take to startup (from the minute you open the app to the minute it actually starts mining) on usual?

**Comment by shainodev** [comment]
also in a previous issue i asked you how to split hashrate, but now i want to know how do i make it so it mines 10% to one address and 90% to other address

**Comment by UnamSanctam** [comment]
>nvm. got it to work, but now im wondering i switched to the moneroocean's xmrig, am i losing any crucial features that u have added? i dont mean the stealth targets or idle mining or any such thing i mean like a important bugfix in xmrig or something that decreases detections alot or am i not?

Only things like automatic core/thread restarts if it ever gets stuck at 0%, some string obfuscation and a few other similar things. Though those are not critical to have.

>also how long does the miners take to startup (from the minute you open the app to the minute it actually starts mining) on usual?

Depends on the computer, since for RandomX it has to generate the dataset. But from the start of the built miner to the injected miner starting takes only really as long as your "Start Delay" is plus a small amount of time to install it if you have that enabled.

>also in a previous issue i asked you how to split hashrate, but now i want to know how do i make it so it mines 10% to one address and 90% to other address

Either you would have to use the donation system in xmrig, or use some pool that supports that like MoneroOcean (where it's entered in the wallet address).

**Comment by shainodev** [comment]
> Either you would have to use the donation system in xmrig, or use some pool that supports that like MoneroOcean (where it's entered in the wallet address).

well i cant do that can i now? if u are not down to give the source, maybe you could add the donation thing for me in your xmrig? or is that too much work?

**Comment by shainodev** [comment]
well i kinda want your version of the xmrig to have it

**Comment by shainodev** [comment]
![image](https://github.com/UnamSanctam/SilentCryptoMiner/assets/77803435/36d2702e-23a3-4b07-b720-c7e87166a296)
is this the correct settings? shouldnt the password have the workername (,ID=Test) instead of worker name textbox?

**Comment by shainodev** [comment]
is there a way to add a wallet ban check to the watchdog

**Comment by UnamSanctam** [comment]
>well i want your version of the xmrig to have it, if thats possible to do

Not at the moment no.

>is this the correct settings? shouldnt the password have the workername (,ID=Test) instead of worker name textbox?

Zergpool only uses the password field, so you should add the `,ID=Test` into the password.

>is there a way to add a wallet ban check to the watchdog

No, wallet bans are fully on the pools side and there's no specific way that they are reported to the user by the pool.

**Comment by shainodev** [comment]
"--pool=stratums://`{data[1]}`.Workername:c=XMR@kawpow.mine.zergpool.com:13638" like this?

**Comment by UnamSanctam** [comment]
It would be `--pool=stratums://{data[1]}:c=XMR,ID=Workername@kawpow.mine.zergpool.com:13638`

**Comment by shainodev** [comment]
> Either you would have to use the donation system in xmrig, or use some pool that supports that like MoneroOcean (where it's entered in the wallet address).

Yes, I use moneroocean. I meant how do I format it WALLET%10%WALLET2, will the wallet2 recieve the 10% or wallet1? also is it possible to see the pool connection i mean wallet from httpdebugger and such?

**Comment by UnamSanctam** [comment]
>Yes, I use moneroocean. I meant how do I format it WALLET%10%WALLET2, will the wallet2 recieve the 10% or wallet1? also is it possible to see the pool connection i mean wallet from httpdebugger and such?

That should be the correct format, and yes wallet2 should receive 10% when it's done like that. And yes since the pool connection is done over the internet then it can be sniffed (though if using SSL then it becomes a bit more difficult), the same as all other internet connections.

**Comment by shainodev** [comment]
How would be the best way to make miner have a devfee without getting the dev fee address banned from a pool? i thought of using 10 xmr addresses and then choosing a random one to use for it but that will just make the ban process slower. maybe is there a way to generate unlimited xmr addresses to use that automatically sends to my other wallet or something?

**Comment by shainodev** [comment]
Prefferably looking for a pool that never bans you, does nanopool never ban or does that depend on the number of miners?

**Comment by Alcinzal** [comment]
> Prefferably looking for a pool that never bans you, does nanopool never ban or does that depend on the number of miners?

I think the two best pools is nanopool and 2miners, haven't heard anyone get banned from those.

**Comment by UnamSanctam** [comment]
As @Alcinzal said, I have never heard of anyone getting banned from nanopool or 2miners either, so those would probably be safe bets.

**Comment by shainodev** [comment]
have you heard of anyone getting banned from moneroocean?

**Comment by UnamSanctam** [comment]
Yes I believe people have gotten banned from MoneroOcean.

**Comment by shainodev** [comment]
is it possible to make it so the commandline supplied to your xmrig.exe is not encrypted?

**Comment by UnamSanctam** [comment]
Yes, my xmrig can take an unencrypted command line, or an encrypted command line, or an encrypted command line through the PEB. All three methods work.

**Comment by shainodev** [comment]
but i tried and it didnt work just taking out the unamlib_encrypt from the builder.cs and leaving {minerid} {argstr.tostring().replace("\"", "\\\"")} and it just insta closes even though with the encryption it works

**Comment by Chimera16** [comment]
@shainodev I also would like to compile and use my own version of xmrig. Is it possible that you share a basic construct of your repository?

**Comment by shainodev** [comment]
> @shainodev I also would like to compile and use my own version of xmrig. Is it possible that you share a basic construct of your repository?

Which one? the XMRig or SilentCryptoMiner?

**Comment by shainodev** [comment]
> Yes, my xmrig can take an unencrypted command line, or an encrypted command line, or an encrypted command line through the PEB. All three methods work.

Why isn't it possible to take an unencrypted command line through the PEB?

**Comment by Chimera16** [comment]
> > @shainodev I also would like to compile and use my own version of xmrig. Is it possible that you share a basic construct of your repository?
> 
> Which one? the XMRig or SilentCryptoMiner?

The XMRig

**Comment by shainodev** [comment]
> The XMRig

What parts exactly do you want me to show/explain?

**Comment by Chimera16** [comment]
> > The XMRig
> 
> What parts exactly do you want me to show/explain?

My goal is to be able to compile my own unique version of XMRig that is compatible with Unams software. It seems like this is also your goal. Is it possible that you share a modified repository of the XMRig that is compatible with Unams software, so I can modify it for myself?

**Comment by shainodev** [comment]
> My goal is to be able to compile my own unique version of XMRig that is compatible with Unams software. It seems like this is also your goal. Is it possible that you share a modified repository of the XMRig that is compatible with Unam's software, so I can modify it for myself?

I can't share my code, but everything you need to make it compatible is here. The only thing you pretty much need is getting the command line from PEB what is in here, and command line arguments parsing for some unique features what I haven't done myself so I have nothing to share on that part.

**Comment by Chimera16** [comment]
Alright I will try it myself. Is the checker able to detect your modified version, so I know that I've done no mistakes?

**Comment by UnamSanctam** [comment]
>Why isn't it possible to take an unencrypted command line through the PEB?

Because if someone is already taking the time to hide it through the PEB then also encrypting it for further stealth is an easy extra step. Only passing it through the PEB and not encrypting it wouldn't make too much sense.

>Alright I will try it myself. Is the checker able to detect your modified version, so I know that I've done no mistakes?

It should be able to detect your version, in your xmrig take the first argument sent to the miner (the miner ID) and create a global mutex with that string.

**Comment by shainodev** [comment]
> Because if someone is already taking the time to hide it through the PEB then also encrypting it for further stealth is an easy extra step. Only passing it through the PEB and not encrypting it wouldn't make too much sense.

Easy step only if it is in C#, I can't easily do it in C++ since I need to add the aes libraries and stuff. For both encryption and decryption.

**Comment by UnamSanctam** [comment]
>Easy step only if it is in C#, I can't easily do it in C++ since I need to add the aes libraries and stuff. For both encryption and decryption.

Are you talking about your version of xmrig? Then you can pass it unencrypted to your xmrig and use it that way without any issue. Though xmrig already has OpenSSL in it so you can just use that for encryption/decryption.

**Comment by shainodev** [comment]
> Are you talking about your version of xmrig? Then you can pass it unencrypted to your xmrig and use it that way without any issue. Though xmrig already has OpenSSL in it so you can just use that for encryption/decryption.

No, I am talking about watchdog, I have made it so the watchdog has the injection and miner resources, and want to change the idle and max cpu & gpu smartly or randomly i havent thought out it yet, and i cant do it without needing to "generate" the commandline again and it has to do it all in c++ so i dont really want to figure out how to add aes in gcc and deal with the bigger file size.

**Comment by Chimera16** [comment]
@shainodev is your version of Xmrig visible in the checker? I have defined the PEB as Unam mentioned. I have also created a global mutex with the first argument that is send to Xmrig. Is that all? Because my Xmrig is still not detected by the checker.

**Comment by UnamSanctam** [comment]
>@shainodev is your version of Xmrig visible in the checker? I have defined the PEB as Unam mentioned. I have also created a global mutex with the first argument that is send to Xmrig. Is that all? Because my Xmrig is still not detected by the checker.

The checker uses the mutex to check if it's running, make sure that your mutex is correct. If you're using the Windows API then the mutex should be `Global\MINERID`

**Comment by Chimera16** [comment]
Hi @UnamSanctam Yes, I have used this:
CreateMutexW(NULL, TRUE, L"Global\\MINERID");
from another comment of you. To be sure: This needs to be placed in my Xmrig build, correct?

**Comment by UnamSanctam** [comment]
Yes, and the MINERID should be the first argument that got sent to the xmrig.

**Comment by Chimera16** [comment]
@shainodev Did you get it to work? My XMrig closes immediately, althrough the minerid should be loaded correctly.
